guides/kfp-rebase-guide.md

```diff
@@ -0,0 +1,197 @@
+    - [2. Clone the `kfp-tekton-bot` fork](#2-clone-the-kfp-tekton-bot-fork)
+  - [Find out What Changed Since the Last KFP Rebase](#find-out-what-changed-since-the-last-kfp-rebase)
+    - [1. Clone the KFP repository](#1-clone-the-kfp-repository)
+    - [2. Get the KFP version numbers we need to compare](#2-get-the-kfp-version-numbers-we-need-to-compare)
+    - [3. Create a `.patch` of the KFP changes since we last merged](#3-create-a-patch-of-the-kfp-changes-since-we-last-merged)
+    - [4. Get the latest changes from `kfp-tekton` `upstream/master`](#4-get-the-latest-changes-from-kfp-tekton-upstream/master)
+    - [5. Create a new branch for the merge](#5-create-a-new-branch-for-the-merge)
+    - [6. Apply the `.patch`](#6-apply-the-patch)
+    - [7. Remove unwanted files, like `OWNERS` in subdirectories](#7-remove-unwanted-files-like-owners-in-subdirectories)
+    - [8. Report number of changed files and the number of rejects](#8-report-number-of-changed-files-and-the-number-of-rejects)
+    - [9. (Optional) Undo `git apply`, reapply `.patch` using IDE or other tools](#9-optional-undo-git-apply-reapply-patch-using-ide-or-other-tools)
+  - [Create a Pull Request with the Latest KFP Changes](#create-a-pull-request-with-the-latest-kfp-changes)
+  - [How to Accept a PR on the Rebase Branch](#how-to-accept-a-pr-on-the-rebase-branch)
+  - [Troubleshooting](#troubleshooting)
+
+<!-- END of ToC generated by running ./tools/mdtoc.sh guides/kfp-rebase-guide.md -->
+
+## Prerequisites
+
+### 1. Generate an GitHub API Token
+
+Generate an GitHub API Token for the [`kfp-tekton-bot` account](https://github.com/kfp-tekton-bot/) 
+with `repo` and `workflow` scope for `git` command line use under
+https://github.com/settings/tokens/new (must be logged in as `kfp-tekton-bot`).
+
+Reach out to @ckadner to get access to the `kfp-tekton-bot` account or to generate
+an API token for you.
+   
+See https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token
+   
+Export the API Token in your terminal:
+```Bash
+export KFP_TEKTON_BOT_API_TOKEN="<your kfp-tekton-bot Github API token>"
+```
+
+<!-- TODO: add some details -->
+
+
+### 2. Clone the `kfp-tekton-bot` fork
+
+Clone the [`kfp-tekton-bot` fork](https://github.com/kfp-tekton-bot/kfp-tekton/)
+of this repository:
+
+```Bash
+git clone https://github.com/kfp-tekton-bot/kfp-tekton.git kfp-tekton-bot
+cd kfp-tekton-bot
+```
+
+
+## Find out What Changed Since the Last KFP Rebase
+
+### 1. Clone the KFP repository
+
+```Bash
+rm -r -f temp/kubeflow/pipelines
+mkdir -p temp/kubeflow/pipelines
+git clone https://github.com/kubeflow/pipelines.git temp/kubeflow/pipelines
+```
+
+
+### 2. Get the KFP version numbers we need to compare
+
+```Bash
+# export KFP_VERSION_OLD=$( grep -E "kfp[=<>]" sdk/python/requirements.in | grep -oE "\d+\.\d+\.\d+" )
+export KFP_VERSION_OLD=$( cat .kfp-rebase-version )
```

Comment:
**NOTE**: @Tomcli this will only work after this PR is merged. 

Until then use:
```Bash
export KFP_VERSION_OLD="1.5.0"
```